pkg/controller/seed-controller-manager/application-secret-cluster-controller/controller.go

```
@@ -0,0 +1,258 @@
+)
+
+type reconciler struct {
+	client              ctrlruntimeclient.Client
+	log                 *zap.SugaredLogger
+	recorder            record.EventRecorder
+	workerName          string
+	namespace           string
+	workerlabelSelector labels.Selector
+}
+
+func Add(
+	ctx context.Context,
+	mgr manager.Manager,
+	log *zap.SugaredLogger,
+
+	numWorkers int,
+	workerName string,
+	namespace string,
+
+) error {
+	workerlabelSelector, err := workerlabel.LabelSelector(workerName)
+	if err != nil {
+		return err
+	}
+
+	r := &reconciler{
+		log:                 log.Named(ControllerName),
+		client:              mgr.GetClient(),
+		workerName:          workerName,
+		recorder:            mgr.GetEventRecorderFor(ControllerName),
+		namespace:           namespace,
+		workerlabelSelector: workerlabelSelector,
+	}
+
+	c, err := controller.New(ControllerName, mgr, controller.Options{Reconciler: r, MaxConcurrentReconciles: numWorkers})
+	if err != nil {
+		return err
+	}
+
+	if err := c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{}, predicateutil.ByAnnotation(applicationsecretsynchronizer.SecretTypeAnnotation, "", false), predicateutil.ByNamespace(r.namespace)); err != nil {
+		return fmt.Errorf("failed to create watch for secrets: %w", err)
+	}
+
+	if err := c.Watch(&source.Kind{Type: &kubermaticv1.Cluster{}}, handler.EnqueueRequestsFromMapFunc(enqueueSecret(ctx, r.client, r.namespace)), workerlabel.Predicates(workerName), noDeleteEventPredicate()); err != nil {
+		return fmt.Errorf("failed to create watch for secrets: %w", err)
+	}
+
+	if err := mgr.GetFieldIndexer().IndexField(ctx, &kubermaticv1.Cluster{}, clusterPauseKey, func(rawObj ctrlruntimeclient.Object) []string {
+		cluster := rawObj.(*kubermaticv1.Cluster)
+		return []string{strconv.FormatBool(cluster.Spec.Pause)}
+	}); err != nil {
+		log.Fatalw("failed to add index on cluster.Spec.Pause: %w", err)
```

Please return this error instead.